src/core/flow.js

```diff
@@ -529,6 +529,18 @@ export function Flow({ moves, phases, endIf, turn, events, plugins }) {
     // Reset activePlayers.
     ctx = { ...ctx, activePlayers: null };
 
+    // Remove player from playerOrder
+    playerID = playerID || state.ctx.currentPlayer;
+
+    if (arg && true === arg.remove) {
+      const playOrder = [...ctx.playOrder].filter(i => i != playerID);
```

Comment:
If you don't also adjust `playOrderPos` here you might leave it out of bounds (when removing the player at the end of `playOrder`).

The following test demonstrates the problem:

```js
  test('test', () => {
    let t = state;
    t = flow.processEvent(t, gameEvent('pass'));
    t = flow.processEvent(t, gameEvent('pass'));
    t = flow.processEvent(t, gameEvent('pass', { remove: true }));
    expect(t.ctx.currentPlayer).toBe('0');   // this fails because currentPlayer is undefined here.
  });
```

Reply:
Good idea. Added my idea about it and added that test as well for posterity. Thanks for the review.